### PR TITLE
Add k8s 1.18

### PR DESCRIPTION
**Pull request by aiyengar2**
This PR adds k8s 1.18.2 to Rancher.

Related Issue: https://github.com/rancher/rancher/issues/25117

### CODE REVIEW

**Comment by mrajashree** [comment]
@aiyengar2 Does the k8s version info map need to be updated with 1.18? https://github.com/rancher/kontainer-driver-metadata/blob/dev-v2.5/rke/k8s_version_info.go#L36

**Comment by prachidamle** [comment]
@aiyengar2 how does this version v1.18.2-rancher1-1 differ from v1.17.5-rancher1-1? 
The format we follow is, first commit is always a "base copy" of the latest version and the next commit you change the images for the new version. This way its easier to know what changed. You can check the examples in the older commits that add new versions. 
Correct me, if that is not the format @kinarashah

**Comment by aiyengar2** [comment]
@mrajashree good catch, forgot to add that. Updating in the latest commit.

@prachidamle that makes sense! Will push out a commit with changes formatted in that way to make it easier to view the differences. Also listed as a comment in the main issue:
https://github.com/rancher/rancher/issues/25117#issuecomment-624915358

File: rke/k8s_version_info.go [hunk 1]
```diff
@@ -115,6 +115,12 @@ func loadK8sVersionInfo() map[string]v3.K8sVersionInfo {
 			MinRancherVersion: "2.3.7-rc0",
 			MinRKEVersion:     "1.0.7",
 		},
+		// The Calico/Canal template in this version use functions that are only available in RKE v1.0.0 and up
+		// This version includes nodelocal dns only available in RKE v1.0.7 and up
+		"v1.18.2-rancher1-1": {
+			MinRancherVersion: "2.3.7-rc0",
```
Comment (kinarashah) [inline rke/k8s_version_info.go:121]: this should be updated, cuz 1.18 is only >=2.4.4

**Comment by aiyengar2** [comment]
> could we also add 1.18 here? https://github.com/rancher/kontainer-driver-metadata/blob/dev-v2.5/rke/k8s_docker_info.go#L14

Updated in the latest commit!

File: rke/k8s_version_info.go (hunk 1 above)
Comment (aiyengar2) [reply rke/k8s_version_info.go]: Updated in the latest push!

**Pull request merged**